networklibrary/src/main/java/com/eis/communication/network/commands/InvitePeer.java

```diff
@@ -0,0 +1,30 @@
+package com.eis.communication.network.commands;
+
+import androidx.annotation.NonNull;
+
+import com.eis.communication.Peer;
+
+/**
+ * Command to Invite a Peer to a network
+ *
+ * @author Marco Cognolato
+ * @author Giovanni Velludo
+ */
+public abstract class InvitePeer<T extends Peer> extends Command {
```

Comment:
Why not a `SendInvitation` command? Because invitations could contain more than just the invited peer, they could contain a message or the network code or whatever. By using the `InvitePeer` with the peer as a parameter you're adding a step to the process of creation of the invitation and you're forcing to have a different command for each type of different invitation (I know there will never be more than one type of invitation, but I like to think bigger and adopt the best practice).

Reply:
well, then this would just be a renaming of a command, right? from `InvitePeer` to `SendInvitation`

Reply:
Nope, I mean `SendInvitation<I extends Invitation>`, so that the invitation can be passed as parameter and already contains what it wants (eg. Peer, a string containing an invitation text, network code, etc.)

Reply:
Oh, so doing it from outside, got it. Will do 👍